sol/providers.bzl

```diff
@@ -5,6 +5,17 @@ SolSourcesInfo = provider(
     fields = {
         "direct_sources": "list of sources provided to this node",
         "transitive_sources": "depset of transitive dependency sources",
+    },
+)
+
+SolRemappingsInfo = provider(
+    doc = """Stores a dictionary of solc remappings.
+    
+    Allows for piping of remappings through a dependency tree of targets. Rules
+    that accept a "remappings" attribute and/or dependencies that provide
+    SolRemappingsInfo SHOULD propagate their union via a SolRemappingsInfo.
```

Comment:
if you use the factory function, then rules should never construct a `SolRemappingsInfo` themselves, which makes it harder for them to get the propagation wrong. You can make the SolRemappingsInfo symbol only available in the private/ folder, then make the factory function the public API instead

Reply:
Done.